docs/core/compatibility/core-libraries/7.0/reflection-invoke-exceptions.md

```diff
@@ -0,0 +1,68 @@
+
+The change to throw <xref:System.NotSupportedException> when a [byref-like type](xref:System.Type.IsByRefLike) is passed to an `Invoke()` API fixes an oversight of the original implementation, which did not throw. The original implementation gave the appearance that `ref struct` types are supported by the `Invoke()` APIs, when they aren't. Since the current `Invoke()` APIs use <xref:System.Object?displayProperty=fullName> for parameter types, and a `ref struct` type can't be boxed to <xref:System.Object?displayProperty=fullName>, it's an unsupported scenario.
+
+## Recommended action
+
+If you don't use <xref:System.Reflection.BindingFlags.DoNotWrapExceptions?displayProperty=nameWithType> when calling `Invoke()`, and you have `catch` statements around the `Invoke()` APIs for exceptions other than <xref:System.Reflection.TargetInvocationException>, consider changing or removing those `catch` statements. The other exceptions will no longer be thrown as a result of the invocation. If, however, you're catching exceptions from argument validation that happens before attempting to invoke the target method, you should keep those `catch` statements. Invalid arguments that are validated before attempting to invoke are thrown without being wrapped with <xref:System.Reflection.TargetInvocationException> and did not change semantics.
+
+Consider using <xref:System.Reflection.BindingFlags.DoNotWrapExceptions?displayProperty=nameWithType> so that <xref:System.Reflection.TargetInvocationException> is never thrown. In this case, the originating exception won't be wrapped by a <xref:System.Reflection.TargetInvocationException>. In most cases, not wrapping the exception improves the chances of the diagnosing the actual issue since not all exception reporting tools display the inner exception. In addition, by using <xref:System.Reflection.BindingFlags.DoNotWrapExceptions?displayProperty=nameWithType>, the same exceptions will be thrown as when calling the method directly (without reflection). This is desirable in most cases, since the choice of whether reflection is used or not can be arbitrary or an implementation detail that does not need to surface to the caller.
+
+In the rare case that you need to pass a default value to a method through reflection that contains a [byref-like](xref:System.Type.IsByRefLike) parameter that's passed "by value", you can add a wrapper method that omits the parameter and calls the target method with a default value for that parameter.
+
+## Affected APIs
+
+- <xref:System.Reflection.MethodBase.Invoke(System.Object,System.Object[])?displayProperty=fullName>
```

Comment:
@steveharter You had listed MethodInfo.Invoke, but I changed that to MethodBase.Invoke. Is that correct? The only MethodInfo.Invoke method I found was for .NET Framework 1.1.